Bottom: b854dbf3454adedfc9b5a4a987834802950b2635
Top:    1e06dcca1ee5c90d14e44cbdb2af2f8f12ef9c06
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-31 01:42:51 +0100

Refresh of drop-in-pcb-gl-code-various-mess

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 0caf5e4..8bd165f 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1049,6 +1049,7 @@ ghid_pinout_preview_expose (GtkWidget *widget,
   if (!gdk_gl_drawable_gl_begin (pGlDrawable, pGlContext)) {
     return FALSE;
   }
+  port->render_priv->in_context = true;
 
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
@@ -1094,6 +1095,7 @@ ghid_pinout_preview_expose (GtkWidget *widget,
     glFlush ();
 
   /* end drawing to current GL-context */
+  port->render_priv->in_context = false;
   gdk_gl_drawable_gl_end (pGlDrawable);
 
   gport->zoom = save_zoom;
@@ -1159,6 +1161,7 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
   if (!gdk_gl_drawable_gl_begin (gldrawable, glcontext)) {
     return NULL;
   }
+  port->render_priv->in_context = true;
 
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
@@ -1203,6 +1206,7 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
   glFlush ();
 
   /* end drawing to current GL-context */
+  port->render_priv->in_context = false;
   gdk_gl_drawable_gl_end (gldrawable);
 //  gdk_gl_context_destroy (glcontext);
